From da3e742132bab6bd6b783dd4a62fecb96ee753e5 Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Mon, 17 Apr 2023 06:42:31 +0000
Subject: [PATCH 37/53] [Huawei] mm/sharepool: Delete the extra spg_id check
 for auto spg_id range

Offering: HULK
hulk inclusion
category: feature
bugzilla: N/A

--------------------------------

The check for spg_id in auto range is not accurate because we check the
spg without the sp_global_sem and the state for the sp_group with that
id would have changed when we actually use it.

Just deleting this check doesn't break the process because we do the
same check in find_or_alloc_sp_group().

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
---
 mm/share_pool.c | 33 +--------------------------------
 1 file changed, 1 insertion(+), 32 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index f0f864302f26..a37e8e92eb3f 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -998,16 +998,6 @@ static struct sp_group *sp_group_get_locked(int tgid, int spg_id)
 	return spg;
 }
 
-static struct sp_group *sp_group_get(int tgid, int spg_id)
-{
-	struct sp_group *spg;
-
-	down_read(&sp_global_sem);
-	spg = sp_group_get_locked(tgid, spg_id);
-	up_read(&sp_global_sem);
-	return spg;
-}
-
 /**
  * mp_sp_group_id_by_pid() - Get the sp_group ID array of a process.
  * @tgid: tgid of target process.
@@ -1348,8 +1338,7 @@ int mg_sp_group_add_task(int tgid, unsigned long prot, int spg_id)
 	check_interrupt_context();
 
 	/* only allow READ, READ | WRITE */
-	if (!((prot == PROT_READ)
-	      || (prot == (PROT_READ | PROT_WRITE)))) {
+	if (!((prot == PROT_READ) || (prot == (PROT_READ | PROT_WRITE)))) {
 		pr_err_ratelimited("prot is invalid 0x%lx\n", prot);
 		return -EINVAL;
 	}
@@ -1359,26 +1348,6 @@ int mg_sp_group_add_task(int tgid, unsigned long prot, int spg_id)
 		return -EINVAL;
 	}
 
-	if (spg_id >= SPG_ID_AUTO_MIN && spg_id <= SPG_ID_AUTO_MAX) {
-		spg = sp_group_get(tgid, spg_id);
-
-		if (!spg) {
-			pr_err_ratelimited("spg %d hasn't been created\n", spg_id);
-			return -EINVAL;
-		}
-
-		down_read(&spg->rw_lock);
-		if (!spg_valid(spg)) {
-			up_read(&spg->rw_lock);
-			pr_err_ratelimited("add group failed, group id %d is dead\n", spg_id);
-			sp_group_put(spg);
-			return -EINVAL;
-		}
-		up_read(&spg->rw_lock);
-
-		sp_group_put(spg);
-	}
-
 	down_write(&sp_global_sem);
 
 	ret = get_task(tgid, &tsk);
-- 
2.17.1

